src/pc/sim_pc.c

```diff
@@ -106,14 +106,16 @@ void func_800E7D08(void) {
     D_800A04EC = 1;
 }
 
-void LoadStageTileset(u8* pTilesetData, s32 y) {
+void LoadStageTileset(u8* pTilesetData, size_t len, s32 y) {
     RECT rect;
     u8* pTilesetDataSrc;
-    s32 i;
+    u32 i;
+    u32 blockCount;
 
     rect.w = 0x20;
     rect.h = 0x80;
-    for (i = 0; i < 0x20; i++) {
+    blockCount = CLAMP_MAX(len >> 13, 0x20);
```

Comment:
Not sure where the right spot is but maybe we should log a warning if we are clamping the length?